pkg/vpa/test_constants.go

```diff
@@ -74,3 +84,12 @@ var testDeployment = &appsv1.Deployment{
 		Name: "test-deploy",
 	},
 }
+
+var testDeploymentExcluded = &appsv1.Deployment{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "test-deploy",
+		Annotations: map[string]string{
+			"goldilocks.fairwinds.com/vpa-opt-out": "true",
```

Comment:
Not necessarily for this PR, but with the update mode being configurable, I think maybe this should be a `vpa-update-mode-override` or something like that. Or maybe a pair of labels/annotations. One for update mode and one to just opt out entirely. 

Looking further below, I am a little confused as to the intent of this particular annotation. The README says `If you do not want a specific Deployment to have a VPA you can annotate the deployment`, but in the reconciler, you just set the update mode to `off`. I think we need a little more clarity around the intent and functionality of this/these annotations for the deployments

Reply:
Yes, this is a good point. I think the purpose behind this opt-out is to opt-out specifically from having the VPA in "auto" mode or having the VPA in an "updater" mode that would actually evict pods and apply recommendations. Perhaps I can make that clearer in the documentation?

Reply:
Clearer documentation is definitely good. I think maybe a different name for it as well.

Perhaps a `goldilocks.fairwinds.com/vpa-mode: off` or something like that. This enables further customization down the road if we want to enable more of this behavior

Reply:
I'll implement this later today

- [ ] Implement more specific control for Deployment annotation

Reply:
"today" turned into Monday, whoops! Let me know what you think of the updated PR.